src/game/entities/entity.ts

```diff
@@ -30,11 +30,45 @@ export interface HP {
   max: number;
 }
 
+// type EquipmentSlotName =
+//   | "head"
+//   | "neck"
+//   | "torso"
+//   | "back"
+//   | "wrist"
+//   | "hand"
+//   | "finger"
+//   | "waist"
+//   | "legs"
+//   | "feet";
+
+interface EquipmentSlot {
+  equipped: Item | null;
+  // name: EquipmentSlotName;
+}
+
+export interface HumanoidBody {
+  arms?: Array<{
+    fingers: Array<EquipmentSlot>;
+    hand: EquipmentSlot;
+    wrist: EquipmentSlot;
+  }>;
+  back: EquipmentSlot;
+  feet: EquipmentSlot;
+  heads: Array<{
+    head: EquipmentSlot;
+    neck: EquipmentSlot;
+  }>;
+  legs: EquipmentSlot;
```

Comment:
Why do you allow multiple heads, but only 1 legs?
Also why are hands part of arms, but feet not part of legs?

Reply:
I thought about this and initially had legs separated, with a child foot for each. I decided in the end that I don't actually want this kind of complexity in the game, I don't think having different boots on different feet or having a 3 legged-orc wearing trousers on two of its legs made sense, so in the context of my game I was happy to just set legs as a single slot for trousers/robes/etc. and feet as a single slot for shoes and boots. This simplifies equipping logic because I don't want to give the player the opportunity to equip boots individually just to make equipping them less tedious, and so if every item which is a pair of boots requires two empty foot slots, it makes sense to just join those slots and only require an empty feet slot.
As for only having a single legs and feet slot, I just don't think having more than two legs or feet on a humanoid body makes the game more fun. I don't know whether I'll allow any other creature to equip stuff yet, but if I do it won't match the equipment slots of a humanoid.

Reply:
tl;dr: I thought about it and decided this is a valid simplification.

Reply:
I agree with that, but then why are feet not inside the legs group?
Or why are hands inside the arms group?
And why is neck a head (saw that elsewhere)

Reply:
Feet don't need to be inside the legs group. If you were to have multiple legs groups, it would be necessary to keep feet inside because there will always be 1 foot to 1 leg, but as there're only ever two legs (as one slot), then we only ever need two feet (as one slot) and nesting is unnecessary.
For hands, they live inside a group called arms. Every arm has a wrist and hand, and each hand has multiple fingers. Because there can only ever be one hand in a single 'arm' group, the fingers don't need to be nested within the hands for the same reason the feet don't need to be nested in the legs above. However, the hands _do_ need to be nested within the arm so that there are always equal numbers of wrists and hands. The same applied for necks, you can't have a neck without a head or a head without a neck, so they live in a group.

Reply:
You also can't have feet without legs... Why is neck and head grouped under heads, but feet not under legs?
Will there actually be different items on different arms? Or different hands (gloves, not weapons)

Reply:
Yes. **You can only have one slot for legs, and only one slot for feet.** No more. Never any more on a humanoid.
You can have any number of heads groups or arms groups, each with their corresponding components. You have to group these components because otherwise you are not guaranteed  to have matching numbers of heads and necks or wrists and hands, which doesn't make sense. You cannot make the simplification I have made with legs/feet with heads/arms because there can be any number of them.

Reply:
There can absolutely be different items on different arms, any one-handed weapon can be equipped in any hand slot. Your point about gloves makes me realise I need another slot on an arm, which would be for gloves, but I will keep gloves as a 'pair' item, meaning it will require two glove slots to be equipped, so a one-armed creature cannot equip gloves.